src/FluentMigrator.Runner.Core/Conventions/DefaultPrimaryKeyNameConvention.cs

```diff
@@ -29,23 +32,24 @@ public class DefaultPrimaryKeyNameConvention : IColumnsConvention
         /// <inheritdoc />
         public IColumnsExpression Apply(IColumnsExpression expression)
         {
+            if (expression.Columns.Count(x => x.IsPrimaryKey) > 1)
+                throw new InvalidOperationException("Error creating table with multiple primary keys.");
+
             foreach (var columnDefinition in expression.Columns)
             {
+                if (string.IsNullOrEmpty(columnDefinition.Name))
```

Comment:
This should be handled through validation logic. There is a separate layer in the framework for validation of expressions.

Reply:
This part of the code is about use column names. Isn't it good to recognize the error as early as possible? To which specific class do you propose to move this check?

Reply:
Why separate the two? Same reason you would separate input validation from a Controller Action method in ASP.NET MVC - it keeps the core business logic focused on the actual problem domain and allows for more precise errors / warnings if done as a ValidationResult rather than throwing an exception. If you throw, you lose Metadata that could be used to build even more nicer error messages, too - similar to what Patrik Svensson's  SpectreConsole does. In general, throwing exceptions for input validation also prevents "railway oriented programming" where the client gets back a object representing success or failure.

That being said, our Validation layer isn't the best. I realize that's frustrating feedback considering I just told you to use it, but it was pretty good 7 years ago - but third party libraries like FluentValidation have made insane leaps forward since then. We currently pattern the existing one of off         System.ComponentModel.DataAnnotations.Validator. I need to delete the areas of the code that still reference CollectErrors, as they have been marked obsolete for awhile. 

But:

1.The interface for adding validation is here but doesn't let you explicitly handle each expression root type - you would have to do type tests and dispatch off a Type-switch (probably to something like FluentValidation): https://github.com/fluentmigrator/fluentmigrator/blob/main/src%2FFluentMigrator.Abstractions%2FValidation%2FIMigrationExpressionValidator.cs
2. The default implementation should be replaced here: https://github.com/fluentmigrator/fluentmigrator/blob/main/src%2FFluentMigrator.Abstractions%2FValidation%2FDefaultMigrationExpressionValidator.cs

3. The CLI tool supports a Validate command. https://github.com/fluentmigrator/fluentmigrator/blob/main/src%2FFluentMigrator.DotNet.Cli%2FCommands%2FValidate.cs
4. Each Model XxxDefinition class and XxxExpression class has a Validate member.